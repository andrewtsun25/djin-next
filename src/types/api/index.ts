import type { HbvResearchPaper } from "./HbvResearchPaper";
import type { Organization } from "./Organization";
import type { HolisticOfficeModule } from "./HolisticOfficeModule";
import type { HolisticOfficeLink } from "./HolisticOfficeLink";
import { HolisticOfficeLinkType } from "./HolisticOfficeLinkType";
import type { MartialArtsStudio } from "./MartialArtsStudio";
import type { MartialArtsStyle } from "./MartialArtsStyle";
import { MartialArtsStyleType } from "./MartialArtsStyleType";

export type {
  HbvResearchPaper,
  Organization,
  HolisticOfficeLink,
  HolisticOfficeModule,
  MartialArtsStudio,
  MartialArtsStyle,
};
export { HolisticOfficeLinkType, MartialArtsStyleType };
