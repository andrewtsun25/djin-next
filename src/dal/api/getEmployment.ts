import { Employment } from "../../types/api";
import {
  createGetterByIdForFirestoreQuery,
  GetterByIdForFirestoreQuery,
} from "../firestore";
import { employmentsCollection } from "../firestore/collections";
import { mapEmploymentDbEntityToEmployment } from "../mapper";

export const getEmployment: GetterByIdForFirestoreQuery<Employment> =
  createGetterByIdForFirestoreQuery(
    employmentsCollection,
    mapEmploymentDbEntityToEmployment,
  );
