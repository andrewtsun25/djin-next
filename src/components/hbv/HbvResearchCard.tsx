import DescriptionIcon from "@mui/icons-material/Description";
import { CardContent, Slide, Typography } from "@mui/material";
import React from "react";

import { BulletPoints } from "../../components/text";
import { HbvResearchPaper } from "../../types/api";
import { DurationWithOrganizationCardHeader } from "../card";
import { FullWidthCardContainer } from "../layout";
import SkillChips from "../text/SkillChips";
import { HbvResearchCardIconLink } from "./styled";

interface HbvResearchCardProps {
  hbvResearchPaper: HbvResearchPaper;
}

export const HbvResearchCard: React.FC<HbvResearchCardProps> = ({
  hbvResearchPaper: {
    organization,
    name,
    startDate,
    endDate,
    description,
    responsibilities,
    skills,
    url,
  },
}: HbvResearchCardProps) => {
  return (
    <Slide direction="up" in mountOnEnter unmountOnExit>
      <FullWidthCardContainer>
        <DurationWithOrganizationCardHeader
          title={name}
          subtitle={organization.name}
          startDate={startDate}
          endDate={endDate}
          logoUrl={organization.logoUrl}
        />
        <CardContent>
          <HbvResearchCardIconLink
            href={url}
            text="Research Paper"
            icon={<DescriptionIcon />}
            target="_blank"
          />
          <Typography mt={2} paragraph>
            {description}
          </Typography>
          <BulletPoints points={responsibilities} />
          <SkillChips skills={skills} />
        </CardContent>
      </FullWidthCardContainer>
    </Slide>
  );
};
