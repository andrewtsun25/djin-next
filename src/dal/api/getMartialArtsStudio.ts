import { MartialArtsStudio } from "../../types/api";
import { MartialArtsStudioDbEntity } from "../../types/db";
import {
  createGetterByIdForFirestoreQuery,
  GetterByIdForFirestoreQuery,
  MapperFunction,
  martialArtsStudiosCollection,
} from "../firestore";

const mapMartialArtsStudioDbEntityToMartialArtsStudio: MapperFunction<
  MartialArtsStudioDbEntity,
  MartialArtsStudio
> = (dbEntity: MartialArtsStudioDbEntity): MartialArtsStudio => {
  const { joinDate, ...rest } = dbEntity;
  return {
    joinDate: joinDate.toDate(),
    ...rest,
  } as MartialArtsStudio;
};

const getMartialArtsStudio: GetterByIdForFirestoreQuery<MartialArtsStudio> =
  createGetterByIdForFirestoreQuery(
    martialArtsStudiosCollection,
    mapMartialArtsStudioDbEntityToMartialArtsStudio
  );

export default getMartialArtsStudio;
