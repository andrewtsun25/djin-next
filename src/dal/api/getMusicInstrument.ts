import { MusicInstrument } from "../../types/api";
import {
  createGetterByIdForFirestoreCollection,
  GetterByIdForFirestoreCollection,
  musicInstrumentsCollection,
} from "../firestore";

const getMusicInstrument: GetterByIdForFirestoreCollection<MusicInstrument> =
  createGetterByIdForFirestoreCollection(musicInstrumentsCollection);

export default getMusicInstrument;
