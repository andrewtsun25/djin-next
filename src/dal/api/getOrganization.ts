import { Organization } from "../../types/api";
import {
  createGetterByIdForFirestoreCollection,
  GetterByIdForFirestoreCollection,
  MapperFunction,
  organizationsCollection,
} from "../firestore";
import { OrganizationDbEntity } from "../../types/db";

const mapOrganizationDbEntityToOrganization: MapperFunction<
  OrganizationDbEntity,
  Organization
> = (dbEntity: OrganizationDbEntity, organizationId: string) => ({
  ...dbEntity,
  id: organizationId,
});

const getOrganization: GetterByIdForFirestoreCollection<Organization> =
  createGetterByIdForFirestoreCollection(
    organizationsCollection,
    mapOrganizationDbEntityToOrganization
  );

export default getOrganization;
