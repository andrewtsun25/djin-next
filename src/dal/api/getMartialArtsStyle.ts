import { MartialArtsStudio, MartialArtsStyle } from "../../types/api";
import { MartialArtsStyleDbEntity } from "../../types/db";
import { isNotNil } from "../../util";
import {
  AsyncMapperFunction,
  createGetterByIdForFirestoreQuery,
  GetterByIdForFirestoreQuery,
  martialArtsStylesCollection,
} from "../firestore";
import getMartialArtsStudio from "./getMartialArtsStudio";

const mapMartialArtsStyleDbEntityToMartialArtsStyle: AsyncMapperFunction<
  MartialArtsStyleDbEntity,
  MartialArtsStyle
> = async (dbEntity: MartialArtsStyleDbEntity): Promise<MartialArtsStyle> => {
  const { studios: studioRefs, ...rest } = dbEntity;
  const studios: MartialArtsStudio[] = (
    await Promise.all(
      studioRefs.map((studioRef) => getMartialArtsStudio(studioRef.id))
    )
  ).filter(isNotNil);
  return {
    studios,
    ...rest,
  } as MartialArtsStyle;
};

const getMartialArtsStyle: GetterByIdForFirestoreQuery<MartialArtsStyle> =
  createGetterByIdForFirestoreQuery(
    martialArtsStylesCollection,
    mapMartialArtsStyleDbEntityToMartialArtsStyle
  );

export default getMartialArtsStyle;
