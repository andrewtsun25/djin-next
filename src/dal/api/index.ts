import getOrganization from "./getOrganization";
import listHbvResearchPapers from "./listHbvResearchPapers";
import listHolisticOfficeModules from "./listHolisticOfficeModules";
import listHolisticOfficeLinks from "./listHolisticOfficeLinks";

export {
  getOrganization,
  listHbvResearchPapers,
  listHolisticOfficeModules,
  listHolisticOfficeLinks,
};
