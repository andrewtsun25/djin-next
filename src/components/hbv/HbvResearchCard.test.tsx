import { render } from "@testing-library/react";

import { HbvResearchPaper, Organization } from "../../types/api";
import HbvResearchCard from "./HbvResearchCard";

describe("HbvResearchCard", () => {
  it("renders", () => {
    const hbvResearchPaper: HbvResearchPaper = {
      name: "Name",
      organization: {
        name: "Organization",
        logoUrl: "https://via.placeholder.com/300.png",
      },
      mediaUrl: "https://media-url.com",
      startDate: new Date("2022-12-25T12:00:00Z"),
      endDate: new Date("2023-01-01T12:00:00Z"),
      url: "https://paper-link.com",
      description: "description",
      responsibilities: ["responsibility1", "responsibility2"],
      skills: ["skill1", "skill2"],
    };
    const hbvResearchCard = render(
      <HbvResearchCard hbvResearchPaper={hbvResearchPaper} />
    );
    expect(hbvResearchCard).toMatchSnapshot();
  });
});
