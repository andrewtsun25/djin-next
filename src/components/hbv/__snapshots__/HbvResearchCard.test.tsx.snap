// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HbvResearchCard renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-x1xzmy-MuiPaper-root-MuiCard-root"
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Name"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://via.placeholder.com/300.png"
              />
            </div>
          </div>
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
            >
              Name
            </span>
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
            >
              Dec 25, 2022 - Jan 1, 2023: Organization
            </span>
          </div>
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1mnlgc6-MuiTypography-root-MuiLink-root"
            href="https://paper-link.com"
            target="_blank"
          >
            <div
              class="MuiBox-root css-70qvj9"
            >
              <div
                class="MuiBox-root css-1ay9vb9"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DescriptionIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm2 16H8v-2h8zm0-4H8v-2h8zm-3-5V3.5L18.5 9z"
                  />
                </svg>
              </div>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Research Paper
              </p>
            </div>
          </a>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-18194e8-MuiTypography-root"
          >
            description
          </p>
          <ul>
            <li
              role="bulletPoint"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                responsibility1
              </p>
            </li>
            <li
              role="bulletPoint"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                responsibility2
              </p>
            </li>
          </ul>
          <div
            class="MuiBox-root css-am63uh"
          >
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1d499cn-MuiChip-root"
              role="skill"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
              >
                skill1
              </span>
            </div>
            <div
              class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1d499cn-MuiChip-root"
              role="skill"
            >
              <span
                class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
              >
                skill2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-x1xzmy-MuiPaper-root-MuiCard-root"
      style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
          >
            <img
              alt="Name"
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://via.placeholder.com/300.png"
            />
          </div>
        </div>
        <div
          class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
          >
            Name
          </span>
          <span
            class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
          >
            Dec 25, 2022 - Jan 1, 2023: Organization
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1mnlgc6-MuiTypography-root-MuiLink-root"
          href="https://paper-link.com"
          target="_blank"
        >
          <div
            class="MuiBox-root css-70qvj9"
          >
            <div
              class="MuiBox-root css-1ay9vb9"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DescriptionIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zm2 16H8v-2h8zm0-4H8v-2h8zm-3-5V3.5L18.5 9z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Research Paper
            </p>
          </div>
        </a>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-18194e8-MuiTypography-root"
        >
          description
        </p>
        <ul>
          <li
            role="bulletPoint"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              responsibility1
            </p>
          </li>
          <li
            role="bulletPoint"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              responsibility2
            </p>
          </li>
        </ul>
        <div
          class="MuiBox-root css-am63uh"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1d499cn-MuiChip-root"
            role="skill"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              skill1
            </span>
          </div>
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1d499cn-MuiChip-root"
            role="skill"
          >
            <span
              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
            >
              skill2
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
